openmc/surface.py

```diff
@@ -2207,148 +2134,64 @@ def get_hexagonal_prism(edge_length=1., orientation='y',
         prism = -top & +bottom & -upper_right & +lower_right & \
                             +lower_left & -upper_left
 
+        if boundary_type == 'periodic':
+            top.periodic_surface = bottom
+            upper_right.periodic_surface = lower_left
+            lower_right.periodic_surface = upper_left
```

Comment:
To avoid possible confusion, I'd recommend adding an except so the issue is understood by the user.